context.go

```diff
@@ -106,11 +106,16 @@ func (c *Context) IsBackup() bool {
 // WatchOnly returns true iff node takes no active part in consensus.
 func (c *Context) WatchOnly() bool { return c.MyIndex < 0 || c.Config.WatchOnly() }
 
-// CountCommitted returns number of received Commit messages not only for the current
-// epoch but also for any other epoch.
-func (c *Context) CountCommitted() (count int) {
+// CountCommittedOrFailed returns the number of nodes that either have sent the Commit
+// message for any epoch or haven't sent any message from the current epoch at all. It
+// is possible for the node to be committed and failed at the same time (consider the
+// node committed at the previous view), thus, we can't return two separate numbers
+// from this method.
```

Comment:
Why can't you adjust `CountFailed()` to look at `c.CommitPayloads[i]` for unavailable nodes? That'd be sufficient, easier and more appropriate.

Reply:
Because `CountFailed()` called separately from `CountCommitted()` expected to count failed nodes (from its name). It may be confusing for the user if `CountFailed()` won't take into account failed node that has committed at the previous view.

Are you sure we need to keep `CountFailed()`? If so, then maybe rename it?

Reply:
Well, a committed node is not a failed one, easy as that. And we have like two places where it's used, both are perfectly fine with this definition of failed.